### PR TITLE
[Debt] Split off mutating sub and roles

### LINKED ISSUE #8289: ♻️ Create separate mutations to edit User sub and roles
## ♻️ Debt/Refactor

The user model is too big - it encompasses multiple use cases. In particular, updateUser mutations do too much - it makes permission checks complicated. But also, it makes caching efficiently difficult. For example, updating a detail of your profile may lead to refreshing of authentication details, which is unnecessary.

## 🕵️ Details

- Create new mutations for editing sub and roles
- remove sub and roles from updateUser mutations
- Use the new mutations on the frontend

## 🙋‍♀️ Proposed Solution

```
input UpdateUserSubInput {
  userId: UUID!
  sub: String!
}

input UpdateUserRolesInput {
  userId: UUID!
  roleAssignmentsInput: RoleAssignmentHasMany!
}

type Mutation {
  updateUserSub(updateUserSubInput: UpdateUserSubInput!): UserAuthInfo
  updateUserRoles(updateUserRolesInput: UpdateUserRolesInput!): UserAuthInfo
  ...
}
```

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] User sub and roles can only be mutated via their own specific mutations, not via updateUserAsAdmin or updateUserAsUser
- [ ] Frontend uses the new mutations ( instead of updateUserAsAdmin )
- [ ] Edit User form gives the sub field its own form and its own submit button 
- [ ] phpunit tests for the new mutations (including testing that they can only be used by expected roles)

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #8289

## 👋 Introduction

Splits mutating sub and roles into two separate mutations and then does some refactoring to support this change. 

## 🕵️ Details

Mutations carved out, testing updated/added.
Operations updated.
User edit page updated to use the two new mutations.

## 🧪 Testing

1. Navigates to edit a user page and update the user's auth data 

## 📸 Screenshot

image

image

### CODE REVIEW

File: apps/web/src/pages/Users/UpdateUserPage/components/UpdateUserSubForm.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          id: "GOCEuh",
+          description:
+            "Message displayed to user when subject field has been updated successfully.",
+        }),
+      );
+    });
+  };
+
+  return (
+    <section>
```
[the remark this answers is not in the document]
Comment (vd1992) [reply apps/web/src/pages/Users/UpdateUserPage/components/UpdateUserSubForm.tsx]: Oh, I didn't even notice how silly it looked with the input bar stretching all the way 😆 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/8352/commits/fee7d501cc919c07e8204c6eead0c6fa9c3daf37

File: apps/web/src/pages/Users/UpdateUserPage/components/UpdateUserSubForm.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            type="text"
+            name="sub"
+            context={intl.formatMessage({
+              defaultMessage:
+                "The 'subject' is a string that uniquely identifies a user's sign in identity.",
+              id: "WLcP98",
+              description:
+                "Additional context describing the purpose of the users's 'subject' field.",
+            })}
+          />
```
[the remark this answers is not in the document]
Comment (vd1992) [reply apps/web/src/pages/Users/UpdateUserPage/components/UpdateUserSubForm.tsx]: Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8352/commits/fee7d501cc919c07e8204c6eead0c6fa9c3daf37

**Review by esizer — APPROVED** [review]
Awesome! ⭐

**Comment by tristan-orourke** [comment]
Does this mean there's now no difference between updateUserAsAdmin and updateUserAsApplicant?

**Comment by vd1992** [comment]
> Does this mean there's now no difference between updateUserAsAdmin and updateUserAsApplicant?

Possibly...?

image

image

File: api/app/Policies/UserPolicy.php [hunk 3]
```diff
@@ -61,22 +61,30 @@ public function create(User $user)
      *
      * @return \Illuminate\Auth\Access\Response|bool
      */
-    public function update(User $user, User $model, array $request = null)
+    public function update(User $user, User $model)
```
Comment (tristan-orourke) [inline api/app/Policies/UserPolicy.php:64]: Yeeeees, this is so much nicer 😁

**Review by tristan-orourke — APPROVED** [review]
Looks good!

**Pull request merged**